vowpalwabbit/sampling.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace exploration
+{
+  struct sample
+  {
+    // if the pdf is not normalized, this is the normalized probability
+    float probability;
+
+    uint32_t index;
+  };
+
+  sample sample_from_pdf(const char* seed, const float* pdf, uint32_t len);
```

Comment:
Is it considered bad these days to use self-delimiting character strings?

Reply:
well, a lot of the functions ask for length, but if you don't have it, it can be real pain (as you need to call strlen...). not sure.